### PR TITLE
Moving rpc to plugin

### PR DESCRIPTION
**Pull request by melanke**
⚠I am not being able to push because of this file: `.github/workflows/dotnetcore.yml` can you help me?!
---
We are moving the RpcServer to a plugin as discussed on this PR: https://github.com/neo-project/neo-plugins/pull/143

### CODE REVIEW

**Comment by erikzhang** [comment]
image

We can't help. You need a rebase.

**Comment by melanke** [comment]
> image
> 
> We can't help. You need a rebase.

done

**Comment by erikzhang** [comment]
I don't understand why we need these `ToPrimitive()`.

**Comment by melanke** [comment]
@erikzhang is to improve the usability of HttpServer controller, you will be able to receive and return primitive types instead of having to transform from/to JObject inside the RPC/Rest method.

**Comment by melanke** [comment]
Are this changes good? May I change all plugins depending on Rpc to use this HttpServer plugin?

File: src/neo/IO/Json/JArray.cs [hunk 1]
```diff
@@ -151,5 +151,14 @@ public override string ToString()
...
         }
+
+        public override object ToPrimitive()
+        {
+            ArrayList list = new ArrayList();
+
+            items.ForEach(o => list.Add(o.ToPrimitive()));
+
+            return list.ToArray();
+        }
```
Comment (Tommo-L) [inline src/neo/IO/Json/JArray.cs]: If these methods are only used in plugins, it is recommended to put them in plugin extension methods.

**Comment by Tommo-L** [comment]
If we move RPC to plugin, I'm not sure will the `Json`  module will be moved too?

**Comment by erikzhang** [comment]
They should be moved too.

**Comment by shargon** [comment]
`Json` module now is used for SmartContracts

File: src/neo/IO/Json/JArray.cs (hunk 1 above)
Comment (melanke) [reply src/neo/IO/Json/JArray.cs]: I can't do this using extension methods because I need to overload the method.

**Comment by melanke** [comment]
But will `neo-types` be a dependency of `neo`?

**Comment by melanke** [comment]
@shargon so what is the point of moving out with neo-types if it will still be inside neo (as dependency)? I believe it will only bring difficulty working with one more project, don't you think?

**Comment by melanke** [comment]
@erikzhang sorry, I don't understand why this PR was closed, should I do something?

**Comment by erikzhang** [comment]
This PR has not progressed for many days, and we need to move forward as soon as possible. So I removed RPC in #1340, and moved it to plugins: neo-project/neo-modules#160, neo-project/neo-modules#163

**Pull request closed without merging**